hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java

```diff
@@ -138,14 +139,32 @@ public Collection<ConsensusNode> getNodes() {
         var totalStake = new AtomicLong(0L);
         var nodes = new TreeSet<ConsensusNode>();
         var nodeStakes = new HashMap<Long, NodeStake>();
+        ConsensusMode consensusMode = mirrorProperties.getConsensusMode();
+        var nodesInAddressBook = addressBook.getEntries().stream().map(AddressBookEntry::getNodeId)
+            .collect(Collectors.toSet());
 
+        var nodeStakeMaxTimestamp = new AtomicLong(0L);
         nodeStakeRepository.findLatest().forEach(nodeStake -> {
-            totalStake.addAndGet(nodeStake.getStake());
+            if (consensusMode == ConsensusMode.EQUAL) {
+              nodeStake.setStake(1L);
+            } else if (consensusMode != ConsensusMode.STAKE_IN_ADDRESS_BOOK ||
+                  nodesInAddressBook.contains(nodeStake.getNodeId())) {
+              totalStake.addAndGet(nodeStake.getStake());
+            }
             nodeStakes.put(nodeStake.getNodeId(), nodeStake);
+            nodeStakeMaxTimestamp.getAndAccumulate(nodeStake.getConsensusTimestamp(), Math::max);
         });
 
         // For partial mirror nodes, node stake will provide a more accurate count
-        long nodeCount = !nodeStakes.isEmpty() ? nodeStakes.size() : addressBook.getNodeCount();
+        long nodeCount = (consensusMode == ConsensusMode.STAKE_IN_ADDRESS_BOOK || nodeStakes.isEmpty()) ?
+            addressBook.getNodeCount() : nodeStakes.size();
+
+        // if only including address book nodes in stake count, warn if any nodes are excluded
+        if (consensusMode == ConsensusMode.STAKE_IN_ADDRESS_BOOK && addressBook.getNodeCount() != nodeStakes.size()) {
```

Comment:
don't warn when node stake is empty

```suggestion
        if (consensusMode == ConsensusMode.STAKE_IN_ADDRESS_BOOK && addressBook.getNodeCount() != nodeStakes.size() && !nodeStakes.isEmpty()) {
```